File: tests/test-data/bad/js/security.js
const jwt = require('jsonwebtoken');
const cors = require('cors');
const express = require('express');
const app = express();

// BAD: js-security-001 — JWT with hardcoded secret
// ruleid: js-security-001
const token = jwt.sign({ userId: 123 }, 'mysecret');

// BAD: js-security-001 — JWT verify with hardcoded secret
// ruleid: js-security-001
jwt.verify(token, 'hardcoded-secret-key');

// BAD: js-security-002 — CORS wildcard
// ruleid: js-security-002
app.use(cors({ origin: '*' }));

// BAD: js-security-003 — TLS verification disabled
// ruleid: js-security-003
process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';

// BAD: js-security-004 — prototype pollution
// ruleid: js-security-004
const config = Object.assign({}, req.body);

File: tests/test-data/bad/go/security.go
package main

import (
	"crypto/tls"
	"net/http"

	"github.com/golang-jwt/jwt"
)

// BAD: go-security-001 — InsecureSkipVerify
// ruleid: go-security-001
func insecureHTTPClient() *http.Client {
	tr := &http.Transport{
		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
	}
	return &http.Client{Transport: tr}
}

// BAD: go-security-002 — hardcoded JWT secret
// ruleid: go-security-002
func parseToken(tokenStr string) (*jwt.Token, error) {
	return jwt.Parse(tokenStr, func(t *jwt.Token) (interface{}, error) {
		return []byte("hardcoded-jwt-secret-key"), nil
	})
}

// BAD: go-config-001 — http.ListenAndServe without TLS
// ruleid: go-config-001
func startServer(mux *http.ServeMux) error {
	return http.ListenAndServe(":8080", mux)
}
